front50-core/src/main/groovy/com/netflix/spinnaker/front50/config/annotations/AnyProviderExceptRedisCondition.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Schibsted ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.config.annotations;
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class AnyProviderExceptRedisCondition implements Condition {
+
+  @Override
+  public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+    Environment env = context.getEnvironment();
+    return serviceEnabled("sql", env)
+        || serviceEnabled("s3", env)
+        || serviceEnabled("oracle", env)
+        || serviceEnabled("gcs", env)
+        || serviceEnabled("azs", env)
+        || serviceEnabled("swift", env);
+  }
+
+  private static boolean serviceEnabled(String s, Environment env) {
+    return env.getProperty(
+        (s.equals("sql") ? "" : "spinnaker.") + s + ".enabled", Boolean.class, false);
+  }
+}
```

Comment:
Clever, I like this a lot. Probably a pattern that should be used elsewhere for our crazy conditional configs (in lieu of making config simpler).

Reply:
I'm actually of the opposite opinion about this...splitting up the strings and joining them in code makes it essentially impossible to search for where config properties are used.  Obviously such a search was never perfect before, but this makes it much much harder to find cases where we depend on `redis.enabled` or `sql.enabled`.

I think at the expense of a bit of repetition it would be much clearer to just pass the full property we're checking into this function rather than join strings like this.

Of course ideally we'd map these onto strongly-typed `@ConfigurationProperties` classes which should probably be the direction we go for anything doing significant work on property classes.

Reply:
This could be rewritten as an AnyNestedCondition:
```
public AnyProviderExceptRedisCondition extends AnyNestedCondition {
  AnyProviderExceptRedisCondition() { super(ConfigurationPhase.PARSE_CONFIGURATION) }

  @ConditionalOnProperty("sql.enabled")
  static class SqlEnabled {}
  
  @ConditionalOnProperty("spinnaker.s3.enabled")
  static class S3Enabled {}

  //... and so on
}
```
Which gives the niceness of properly referencing the properties we care about via `@ConditionalOnProperty` and also the niceness of not having insane-o SpEL conditions

Reply:
Yeah, that looks better. But it should probably been a part of the original PR, not this cherry pick/backport into the 1.20 branch... I suggest not doing anything here, and rather create a new PR to make the annotation cleaner.

Reply:
I think `AnyNestedCondition` is a much better path... and this PR should probably be updated to use it, seeing how this PR is introducing the annotation to begin with, I don't know why we'd accept this knowing we want to refactor it.

Reply:
Sorry, I’m not following you here. The annotation was introduced in #839 and is merged to master. This is a backport of that PR.
Do you mean to create a new PR that refractors the annotation, and when that is merged to master, include that as well in this PR?

Reply:
Oh sorry, I missed that this is a cherry pick. nvm

Reply:
https://github.com/spinnaker/front50/pull/858